x-pack/plugins/enterprise_search/public/applications/enterprise_search_content/components/search_index/connector/connector_configuration_form.tsx

```diff
@@ -42,12 +59,20 @@ export const ConnectorConfigurationForm = () => {
       component="form"
     >
       {localConfigView.map((configEntry) => {
-        const { key, label } = configEntry;
+        const { depends_on: dependencies, key, label } = configEntry;
+        const hasDependencies = dependencies.length > 0;
 
-        return (
-          <EuiFormRow label={label ?? ''} key={key}>
+        return dependenciesSatisfied(dependencies, dependencyLookup) ? (
+          <EuiFormRow
+            label={label ?? ''}
+            key={key}
+            className={hasDependencies ? 'dependency' : ''}
+            style={hasDependencies ? { backgroundColor: subduedBackground } : {}}
```

Comment:
If this needs CSS, we should apply it via the class in that CSS file, not directly on the style prop